geometry/shapes.py

```diff
@@ -432,3 +432,45 @@ def intersects(obj1, obj2):
             return False
 
     return True
+
+
+class Triangle(Polyhedron):
+    def __init__(self, *args):
+        super(Triangle, self).__init__()
+
+        self._vertices = []
+        input_shape = np.array(args[0]).shape
+
+        if len(args) != 3:
+            raise ValueError("Triangle must have three vertices")
+
+        # Verify that all the input vertices are of dimension 2 or 3. If they are of dimension 2 store them in 3D.
+        for a in args:
+            pt = np.array(a)
+
+            if pt.shape != input_shape:
+                raise ValueError("Triangle base vertices do not all have the same dimension")
+
+            if pt.shape == (2,):
+                pt = np.array((pt[0], pt[1], 0.0, 1.0))
+            elif pt.shape == (3,):
+                pt = np.array((pt[0], pt[1], pt[2], 1.0))
+            elif pt.shape == (4,):
+                pt = utils.make_projective_point(pt)
+            else:
+                raise ValueError("Prism cannot be constructed from non 2d, 3d or projective 3d points. Got %s" % pt)
```

Comment:
Change the error message here. It's a triangle, not a prism.

Also, since it looks like this conversion from 2d to 3d is happening everywhere, why not make it a utility function in `utils.py`? Then update the code everywhere to use it. It'll be much less of a pain in the ass to read. 

In general, don't copy paste code. If you're doing something twice or more, make it a function.